02-Architecture Constraints.md

```diff
@@ -1,3 +1,17 @@
 # Architecture Constraints
 
+An architecture constraint is any requirement that constrains software developers in their freedom of design and implementation decisions or decision about the development process. These constraints sometimes go beyond individual systems and are valid for whole organizations.
 
+## Design and Development constraints
+
+Technical constraints that apply to the design and development of the system.
+
+## [Lua](https://en.wikipedia.org/wiki/Wikipedia:Lua)
+
+Lua is a programming language that is available via the [Scribunto MediaWiki extension](https://www.mediawiki.org/wiki/Extension:Scribunto) on the English Wikipedia. Lua code can be embedded into [wiki templates](https://en.wikipedia.org/wiki/Help:A_quick_guide_to_templates). It's often used in templates for [Infoboxes](https://en.wikipedia.org/wiki/Help:Infobox) on Wikipedias.
+
+The WikibaseClient extension provides a Lua Scribunto interface that implements functions to access data from the Wikibase repository. Lua modules and wiki templates can invoke these functions.
+
+## [Wikitext](https://en.wikipedia.org/wiki/Help:Wikitext)
+
+Wikitext, also known as Wiki markup or Wikicode, consists of the syntax and keywords used by the MediaWiki software to format a page. 
```

Comment:
I think the gist of what makes Wikitext a constraint for us is: "All of a wiki article's content is written in Wikitext, so any Wikibase data we want to be embedded in the article content must go through it.". I added a sentence to convey that, but feel like there's a better way to phrase it.

```suggestion
Wikitext, also known as Wiki markup or Wikicode, consists of the syntax and keywords used by the MediaWiki software to format a page. It's what editors work with to create and edit wiki articles. Any Wikibase entity data displayed as part of the article content is loaded either by a [Lua function](# Lua) or a [ParserFunction](https://www.mediawiki.org/wiki/Help:Extension:ParserFunctions) invoked through Wikitext.
```

Reply:
I think "is loaded" should be "should be loadable" to make this more constrainty?